src/core/inlines.js

```diff
@@ -51,8 +52,21 @@ function inlineRFC2119Matches(matched) {
 function inlineRefMatches(matched) {
   // slices "[[[" at the beginning and "]]]" at the end
   const ref = matched.slice(3, -3).trim();
-  const nodeElement = hyperHTML`<a data-cite="${ref}"></a>`;
-  return nodeElement;
+  if (ref.startsWith("#")) {
+    if (!document.querySelector(ref)) {
```

Comment:
Maybe return early `if (document.querySelector(ref))` and avoid larger nested chunk?

Reply:
That may cause too many queries even for non-id links. But then how about:

```
if (!ref.startsWith("#")) {
  // return link with data-cite
}
// more codes to return link with href

Reply:
good suggestion